config/initializers/omniauth.rb

```diff
@@ -1,64 +1,41 @@
+# OKTA_CLIENT_ID is defined, the app will fail to boot
+#
 Rails.application.config.middleware.use OmniAuth::Builder do
-  # Default usage, this will give you offline access and a refresh token
-  # using default scopes 'email' and 'profile'
-  #
-  redirect_uri = "#{host}/auth/google_oauth2/callback"
-
-  options = {
-    scope:        'email,profile,calendar',
-    redirect_uri: redirect_uri,
-    setup: ->(env) do
-      env['omniauth.strategy'].options['token_params'] = {
-        redirect_uri: redirect_uri
-      }
+  raise 'Please configure either Google or SAML Authentication.' unless ENV['GOOGLE_CLIENT_ID'] || ENV['SAML_ISSUER']
+
+  if ENV['GOOGLE_CLIENT_ID']
+    # Default usage, this will give you offline access and a refresh token
+    # using default scopes 'email' and 'profile'
+
+    redirect_uri = "#{host}/auth/google_oauth2/callback"
+
+    options = {
+      scope:        'email,profile,calendar',
+      redirect_uri: redirect_uri,
+      setup: ->(env) do
+        env['omniauth.strategy'].options['token_params'] = {
+          redirect_uri: redirect_uri
+        }
+      end
+    }
+
+    if domain = ENV['OAUTH_DOMAIN']
+      options.merge!(hd: domain)
     end
-  }
 
-  if domain = ENV['OAUTH_DOMAIN']
-    options.merge!(hd: domain)
+    provider :google_oauth2, ENV.fetch('GOOGLE_CLIENT_ID'), ENV.fetch('GOOGLE_CLIENT_SECRET'), options
   end
 
-  provider :google_oauth2, ENV.fetch('GOOGLE_CLIENT_ID'), ENV.fetch('GOOGLE_CLIENT_SECRET'), options
-
-  # Manual setup for offline access with a refresh token.
-  # The prompt must be set to 'consent'
-  #
-  # provider :google_oauth2, ENV['GOOGLE_KEY'], ENV['GOOGLE_SECRET'], {
-  #   :access_type => 'offline',
-  #   :prompt => 'consent'
-  # }
-
-  # Custom scope supporting youtube. If you are customizing scopes, remember
-  # to include the default scopes 'email' and 'profile'
-  #
-  # provider :google_oauth2, ENV['GOOGLE_KEY'], ENV['GOOGLE_SECRET'], {
-  #   :scope => 'http://gdata.youtube.com,email,profile,plus.me'
-  # }
-
-  # Custom scope for users only using Google for account creation/auth and do not require a refresh token.
-  #
-  # provider :google_oauth2, ENV['GOOGLE_KEY'], ENV['GOOGLE_SECRET'], {
-  #   :access_type => 'online',
-  #   :prompt => ''
-  # }
-
-  # To include information about people in your circles you must include the 'plus.login' scope.
-  #
-  # provider :google_oauth2, ENV['GOOGLE_KEY'], ENV['GOOGLE_SECRET'], {
-  #   :skip_friends => false,
-  #   :scope => "email,profile,plus.login"
-  # }
-
-  # If you need to acquire whether user picture is a default one or uploaded by user.
-  #
-  # provider :google_oauth2, ENV['GOOGLE_KEY'], ENV['GOOGLE_SECRET'], {
-  #   :skip_image_info => false
-  # }
+  if ENV['SAML_ISSUER']
+    provider :saml,
+    :issuer                             => ENV['SAML_ISSUER'],
+    :idp_sso_target_url                 => ENV['SAML_IDP_SSO_TARGET_URL'],
+    :idp_cert                           => ENV['SAML_IDP_CERT']
```

Comment:
these lines should be indented, right? and we probably don't need so much space between key and value. and you can just use `key: value` instead of `:key => value`

Reply:
should these be `ENV.fetch` calls so they blow up if the config is not set?